Every translatable string is now in US_en.json - Added IT_it.json (in Italian)

Montelion opened the pull request:
With this pull request, I'm trying to make the website translatable. I made every translatable word/sentence refer to the localized .json in /locales.

JipFr commented on locales/US_en.json:
```diff
@@ -1,23 +1,60 @@
+	},
+	"faq": {
+		"title": "FAQ"
+		"q1": "What is Pretendo?",
```
Looks like you’re missing a comma

Montelion replied:
Fixed!

JipFr commented on locales/US_en.json:
```diff
@@ -1,23 +1,60 @@
+		"title": "Cemu",
+		"text": "Pretendo is currently not usable by the general public. When the network becomes public, this will contain steps on connecting to the network using Cemu"
+	},
+	"faq": {
```
Maybe make an array of questions instead of assigning aN values?

Montelion replied:
Do you mean this?

```
{ "faq": [ 
    { "question" : "Here I write the first question", "answer" : "And here I write the first answer" } , 
    { "question" : "Same thing but it's the first question", "answer" : "Again, same thing but I write the second answer" } 
] }
```

Do I have to change anything in the FAQ page?
BTW thanks for the help!

JipFr replied:
I’d probably add a questions field to faq so that you can still add the translations for the general title.

However, maybe wait with this a bit. I’ve asked the  other web people what they think (array or your current setup?) and I’ll let you know what they say

Montelion replied:
Ok. Thank you very much!

Montelion replied:
Or did you mean something like this?

```
{
all the other stuff

          "faq": {
                      "questions":[ "Question 1", "Question 2", "Question 3" ],
                      "answers":[ "Answer 1", "Answer 2", "Answer 3" ]
}
}
```
and in the faq page I use `locale.faq.questions[0]` when I want to insert question 1, and `locale.faq.answers[0]` to insert answer 1.
I'm sorry for my dumb questions, I'm fairly new to JSON, and javascript in general.
Again, thanks for your help!

JipFr replied:
I was thinking ```JSON
"faq": {
  "title": "FAQ",
  "questions": [
    {
      "question": "What's 9+10?",
      "answer": "21"
    },
    {
      "question": "What's 9+10?",
      "answer": "21"
    }
  ]
}``` and then forEach it. However, beforer making such a decision I think the other web people should get involved

Montelion replied:
Ok. Thanks!

RedDuckss replied:
I agree with @JipFr on this. Maybe this could be applied to the credits as well? It feels a bit odd using the developer name as the key. Using an array and looping would allow us to add or remove a credit just in the translation file and not have to update the actual markup

Montelion replied:
Hi, everyone. Thanks for all your support. So... I will put the FAQ and the credits in an array.
BTW for the credits would this be ok?

```
"credits": {
	"caption": "The people responsible",
	"title": "Credits",
	"people": [
		{
		"name": "Jonathan Barrow (RedDucks)",
		"caption": "Project owner and lead developer"
		"link": "https://github.com/RedDuckss"
		"picture": "https://github.com/redduckss.png"
		},
		etc.
		]
```

But I'm afraid I won't be able to edit the home.hbs and the faq.hbs, because... I don't know how to code. I searched on the internet for a guide, but I'm lost. I'm sorry. 😓

I will be happy to put the FAQ and the credits in an array, but I can't do more than that.

JipFr replied:
https://handlebarsjs.com/guide/builtin-helpers.html#each this should help you with that ;)

Montelion replied:
Thanks!

JipFr commented on locales/US_en.json:
```diff
@@ -1,23 +1,60 @@
+		"q4": "Does Pretendo work on Cemu/emulators?",
+		"a4": "Pretendo is designed for the WiiU and (eventually) 3DS. At this time the only emulator for these consoles with real online capability is Cemu. Cemu does not officially support custom servers, but it will still be possible to use Pretendo with Cemu
+								<br>
+								See the <a href="/cemu">Cemu</a> page for more details",
```
You’re accidentally closing your JSON tag here

Montelion replied:
Fixed!

JipFr commented on locales/US_en.json:
```diff
@@ -1,23 +1,60 @@
+		"q6": "If I am account/console banned on Nintendo Network, will I stay banned when using Pretendo?",
+		"a6": "Any account/console bans only exist server-side. This means any existing bans would not carry over to Pretendo. However Pretendo still reserves the right to issue a ban for our network if we see fit",
+		"q7": "Will Pretendo support the Wii/Switch?",
+		"a7": "The Wii already has custom servers provided by <a href="https://wiimmfi.de/" target="_blank">Wiimmfi</a>. We currently have no plans on supporting the Switch, as not only is it a completely different platform with a different online network, but Switch online is a paid service"
```
You’re closing the JSON here as well

Montelion replied:
Fixed!

RedDuckss commented:
There are some strings which honestly don't need a translation since they will be the same in any language. Like the Pretendo branding and the Cemu branding. These are titles of software, they will be the same no matter the language

Montelion commented:
> There are some strings which honestly don't need a translation since they will be the same in any language. Like the Pretendo branding and the Cemu branding. These are titles of software, they will be the same no matter the language

Ok. Do I have to remove the ones that were there before my edits? (e.g.

```
"nav": {
	"brand": "Pretendo"
	},
"home": {
	"header": {
		"title": "Pretendo",
```

)

RedDuckss commented:
Yes you can remove those from the language file and just hard code them

Montelion commented:
Ok, so...everything should be ready. Where do I need to put handlebars?

JipFr commented:
> Ok, so... **`theoretically`** everything should be ready. Where do I need to put handlebars?

How do you mean?

Montelion commented:
> > Ok, so... **`theoretically`** everything should be ready. Where do I need to put handlebars?
> 
> How do you mean?

I used this: https://handlebarsjs.com/guide/builtin-helpers.html#each. Do I need to add anything to the code or is it ready?

JipFr commented:
I've made a small few patches, including the following

- I've improved some of your indentation, both in the hbs and the JSON
- I've added a US_en locale string for "connecting"
- I've fixed an error in the JSON, causing the home page not to load.

Now, it looks good to me. We'll need to wait @RedDuckss's judgement, but I think it's good now

Montelion commented:
> I've made a small few patches, including the following
> 
> * I've improved some of your indentation, both in the hbs and the JSON
> * I've added a US_en locale string for "connecting"
> * I've fixed an error in the JSON, causing the home page not to load.
> 
> Now, it looks good to me. We'll need to wait @RedDuckss's judgement, but I think it's good now

Ok. Great! Thanks!

Montelion commented:
I added the Italian translation. Now everything should **actually** be ready.

RedDuckss commented:
Looks good to me! Can I get @JipFr to take one final look and make sure everything is still functional? After that I'll be happy to merge

JipFr commented:
Will do!

JipFr commented:
Looks good to me, everything still works!

However, there is currently no way to switch between locales. It's currently hard-coded as `const locale = utilHelper.getLocale('US', 'en');`. Is this a feature that'll come with the account service?

Montelion commented:
I just noticed this:
![image](https://user-images.githubusercontent.com/49426949/80843093-a6d52b00-8c03-11ea-902c-e836bd6d41ba.png)
Is this something not to worry about, or is this a problem?

JipFr commented:
Oh I missed that. Yeah that’s an issue.

@RedDuckss should we add a Markdown converter and parse new values or just allow HTML in the locales?

JipFr commented:
Alright, you can fix that without supporting MD. You can fix it you replace `{{ localeVar }}` with `{{{ localeVar }}}`

Montelion commented:
Are you talking about this part?
```
			<section class="bigCardWrapper">
				<div class="bigCard">
					<h1 class="txt-title">{{ locale.faq.title }}</h1>
					<div class="faqs">
						{{#each locale.faq.qna}}
						<details>
							<summary>{{question}}</summary>
							<p class="text">{{answer}}</p>
						</details>
						{{/each}}
					</div>
				</div>
			</section>
```
I tried doing as you said for `{{ locale.faq.title }}` and `{{/each}}`, but `Something broke!`, same with `{{question}}` and `{{answer}}`. I also tried with all of them together, but still nothing.

JipFr commented:
Don’t do it for the each, only for the question and answer

Montelion commented:
I did it before, it doesn't work. The output from the teminal is
```
[0:46:17] [ERROR]: Error: Lexical error on line 21. Unrecognized text.
...<p class="text">{{{{answer}}}}</p>
-----------------------^
    at Parser.parseError (C:\Users\matte\Downloads\Pretendo_website-master\Pretendo_website-master\node_modules\handlebars\dist\cjs\handlebars\compiler\parser.js:267:19)
    at Object.parseError (C:\Users\matte\Downloads\Pretendo_website-master\Pretendo_website-master\node_modules\handlebars\dist\cjs\handlebars\compiler\parser.js:394:36)
    at Object.next (C:\Users\matte\Downloads\Pretendo_website-master\Pretendo_website-master\node_modules\handlebars\dist\cjs\handlebars\compiler\parser.js:522:33)
    at Object.lex (C:\Users\matte\Downloads\Pretendo_website-master\Pretendo_website-master\node_modules\handlebars\dist\cjs\handlebars\compiler\parser.js:526:30)
    at lex (C:\Users\matte\Downloads\Pretendo_website-master\Pretendo_website-master\node_modules\handlebars\dist\cjs\handlebars\compiler\parser.js:297:36)
    at Parser.parse (C:\Users\matte\Downloads\Pretendo_website-master\Pretendo_website-master\node_modules\handlebars\dist\cjs\handlebars\compiler\parser.js:320:34)
    at parseWithoutProcessing (C:\Users\matte\Downloads\Pretendo_website-master\Pretendo_website-master\node_modules\handlebars\dist\cjs\handlebars\compiler\base.js:46:33)
    at HandlebarsEnvironment.parse (C:\Users\matte\Downloads\Pretendo_website-master\Pretendo_website-master\node_modules\handlebars\dist\cjs\handlebars\compiler\base.js:52:13)
    at compileInput (C:\Users\matte\Downloads\Pretendo_website-master\Pretendo_website-master\node_modules\handlebars\dist\cjs\handlebars\compiler\compiler.js:508:19)
    at ret (C:\Users\matte\Downloads\Pretendo_website-master\Pretendo_website-master\node_modules\handlebars\dist\cjs\handlebars\compiler\compiler.js:517:18)
```

JipFr commented:
Use 3 brackets on each side, not 4. That should make it work!

Montelion commented:
Whoops. Yeah, it works with 3!

JipFr commented:
I’ll check again tomorrow and then we can merge!

JipFr commented:
LGTM! @RedDuckss let’s get this merged!

Montelion commented:
Ok. Looks good to me too, I changed a translation that didn't sound right, but now everything's ready!
